### PR TITLE
[Flow EVM] fix the issue with the new Geth tracing

### PR DESCRIPTION
**Pull request by ramtinms**
This PR fixes nil pointer problems in case of tx tracing is tracing a invalid transaction.

### CODE REVIEW

**Review by peterargue — APPROVED** [review]

File: fvm/evm/emulator/emulator.go [hunk 1]
```diff
@@ -121,11 +121,16 @@ func (bl *BlockView) DirectCall(call *types.DirectCall) (res *types.Result, err
...
-			defer func() {
-				proc.evm.Config.Tracer.OnTxEnd(res.Receipt(), err)
-			}()
-		}
+		defer func() {
+			if proc.evm.Config.Tracer.OnTxEnd != nil {
+				receipt := &gethTypes.Receipt{}
+				if res != nil {
+					receipt = res.Receipt()
+				}
```
Comment (turbolent) [inline fvm/evm/emulator/emulator.go:129]: Minor: maybe avoid extra allocation in happy case. Same below

```suggestion
				var receipt *gethTypes.Receipt
				if res != nil {
					receipt = res.Receipt()
				} else {
					receipt = &gethTypes.Receipt{}
				}
```

**Review by turbolent — APPROVED** [review]
Just to clarify: The bug is that `res` can be `nil`, so handle it by using an empty receipt instead?

Not familiar with Geth and tracing, but the bugfix for the nil pointer looks good! 👍

File: fvm/evm/emulator/emulator.go (hunk 1 above)
Comment (ramtinms) [reply fvm/evm/emulator/emulator.go:129]: I was not sure the underlying layer could handle the not allocated case.

Comment (turbolent) [reply fvm/evm/emulator/emulator.go:129]: Yeah, allocating a new value in the unhappy path seems reasonable if we don't know if passing down nil is valid. Maybe we can avoid the allocation in the happy path though

**Pull request merged**